### PR TITLE
Autoload ProxyAPI files from /app/services

### PR DESCRIPTION
**Pull request by stejskalleos**

### CODE REVIEW

File: app/models/concerns/orchestration/dhcp.rb [hunk 1]
```diff
@@ -89,7 +89,7 @@ def boot_server
   end
 
   def boot_server_or_proxy_hostname
-    return tftp.bootServer if tftp.bootServer.present?
+    return tftp.boot_server if tftp.boot_server.present?
```
Comment (ezr-ondrej) [inline app/models/concerns/orchestration/dhcp.rb]: we could be more efficient while already touching this:
```suggestion
    return tftp.boot_server.presence
```

File: app/services/proxy_api/tftp.rb [hunk 2]
```diff
@@ -39,7 +39,7 @@ def fetch_boot_file(args)
     end
 
     # returns the TFTP boot server for this proxy
-    def bootServer
+    def boot_server
```
Comment (ezr-ondrej) [inline app/services/proxy_api/tftp.rb]: I'm not sure if this is not used anywhere, it is a public API, so it definetelly could.
So in regards to https://community.theforeman.org/t/how-to-track-developer-changes/26523/5 I'd just drop a notice about this on discoures (unfortunatelly we do not have specialized section for this yet).

**Review by ezr-ondrej — COMMENTED** [review]
Did you change the `boot_server` just because you've seen it?
I know it is hard to resist this, but might be better unless we have incentive to do so.
Don't get me wrong, I like it, I don't agree with the initial reason for it (the one I suspect), it is just it might break stuff unnecessarily :)

File: app/models/concerns/orchestration/dhcp.rb (hunk 1 above)
Comment (ekohl) [reply app/models/concerns/orchestration/dhcp.rb]: No, that would be different. This is an early return with an if. Your suggestion would make the (deprecated) code below unreachable. We haven't made a formal deprecation statement about that. Please keep that for a separate PR.

**Comment by stejskalleos** [comment]
> Did you change the boot_server just because you've seen it?

I was trying to make rubocop happy :D But if I think about it it would be safer to just update `.rubocop_todo.yml` and keep the old naming.

Anyway PR rebased & updated, ready for review

**Review by ezr-ondrej — APPROVED** [review]
Thanks @stejskalleos ! This feels safer :)

**Pull request merged**